### PR TITLE
AMD ROCm support

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
- [x] Adapt `setup.py`
- [x] Adapt `build.yaml` workflow

### CODE REVIEW

**Comment by casper-hansen** [comment]
I have no AMD ROCm GPUs and no way of testing this. Is it correctly understood that it's exclusively ExLlama kernels that will work with AMD GPUs on ROCm 5 (but not 6)?

**Comment by IlyasMoutawwakil** [comment]
I tested on ROCm 5.6.1
@seungrokj do [these changes](https://github.com/AutoGPTQ/AutoGPTQ/pull/515) build on both 5 and 6 ?
If so I can add them in this PR.

**Comment by seungrokj** [comment]
> I tested on ROCm 5.6.1 @seungrokj do [these changes](https://github.com/AutoGPTQ/AutoGPTQ/pull/515) build on both 5 and 6 without impacting performance ? If so I can add them in this PR.

@IlyasMoutawwakil  I just checked https://github.com/AutoGPTQ/AutoGPTQ/pull/515 in rocm5.7 env. It is compiled and there should be no perf issue because they are basically same syntaxes. (Somehow the old syntax is not compatible with rocm6.0 though)

**Comment by IlyasMoutawwakil** [comment]
@casper-hansen for the other part of your question, yes for now Exllama is going to be the only way to run AWQ model on AMD GPUs. Untill we get a hipifiable, ROCm native or performant triton GEMM.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -113,9 +113,108 @@ jobs:
...
+          elif [[ "${{ matrix.rocm }}" == "5.7.1" ]]; then
+            export ROCM_DL_FILE=amdgpu-install_5.7.50701-1_all.deb
+          else
+            echo Unknown rocm version
+            exit 1
+          fi
+
+          curl -O https://repo.radeon.com/amdgpu-install/${{ matrix.rocm }}/ubuntu/focal/$ROCM_DL_FILE
+          sudo dpkg -i $ROCM_DL_FILE
+          sudo DEBIAN_FRONTEND=noninteractive amdgpu-install --usecase=rocm --no-dkms --no-32 -y
```
Comment (casper-hansen) [inline .github/workflows/build.yaml:188]: I see a lot of Linux commands. Should this run on the Windows builder from GitHub? I believe the reason the CUDA build is implemented in PowerShell is that it's cross-platform

Comment (IlyasMoutawwakil) [reply .github/workflows/build.yaml:202]: No only Linux, I made a mistake while trying to copy from this workflow, https://github.com/AutoGPTQ/AutoGPTQ/blob/main/.github/workflows/build_wheels_rocm.yml

**Pull request merged**